openassessment/xblock/team_mixin.py

```diff
@@ -0,0 +1,78 @@
+""" Mixin to encapsulate teams-related behavior """
+import logging
+
+from django.utils.functional import cached_property
+from django.core.exceptions import ObjectDoesNotExist
+from xblock.exceptions import NoSuchServiceError
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class TeamMixin(object):
+    """Team Mixin introducing all teams-related functionality."""
+
+    STAFF_OR_PREVIEW_INFO = {
+        'team_name': '< TEAM NAME >',
+        'team_usernames': ['< USERNAMES >'],
+        'team_url': '#TEAM_URL',
+    }
+    @cached_property
+    def teams_service(self):
+        try:
+            return self.runtime.service(self, 'teams')
+        except NoSuchServiceError:
+            logger.error(u'{}: Teams service unavailable'.format(self.location))
+
+    @cached_property
+    def team(self):
+        """
+        For the user and course_id associated with this ORA block, returns the user's
+        CourseTeam, or None if the user is not a member of a team in this course.
+        Raises:
+            - NoSuchServiceError if the teams service is unavailable
+            - ObjectDoesNotExist if the user associated with `anonymous_user_id`
+                                    can not be found
+        """
+        anonymous_user_id = self.get_anonymous_user_id_from_xmodule_runtime()
+        user = self.get_real_user(anonymous_user_id)
+        if not user:
+            logger.error(u'{}: User lookup for anonymous_user_id {} failed'.format(self.location, anonymous_user_id))
+            raise ObjectDoesNotExist()
+        team = self.teams_service.get_team(user, self.course_id)
+        return team
+
+    def has_team(self):
+        """
+        returns true if the student is on a team, false if the student is not on a team
+        or if an exception is raised while looking up team
+        """
+        try:
+            team = self.team
+        except ObjectDoesNotExist:
+            return False
+        return bool(team)
+
+    def should_hide_team_ora(self):
```

Comment:
I think this might read better if we just had something like a `valid_access_to_team_assessment` function that combines this and `teams_enabled`. That way the `if` checks would say "if we are trying an invalid access to a team assessment, don't show it"

Reply:
... but just a stylistic thought. Feel free to use or disregard.

Reply:
No, I think that you're right, the broader description of what the function does is a better name than this specific use case of the logic